### PR TITLE
Fix errors found by parfait static analyser.

### PR DESCRIPTION
**Pull request by slontis**
##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Review by levitte — APPROVED** [review]

File: providers/implementations/macs/cmac_prov.c [hunk 1]
```diff
@@ -87,6 +87,8 @@ static void *cmac_dup(void *vsrc)
         return NULL;
 
     dst = cmac_new(src->provctx);
+    if (dst == NULL)
+        return NULL;
```
Comment (levitte) [inline providers/implementations/macs/cmac_prov.c:91]: Side note: we really need to add more error reports...

Comment (paulidale) [reply providers/implementations/macs/cmac_prov.c:91]: At what point should be start adding them?

_Now_ seems like a good answer...

**Review by paulidale — APPROVED** [review]

File: providers/implementations/macs/cmac_prov.c (hunk 1 above)
Comment (slontis) [reply providers/implementations/macs/cmac_prov.c:91]: You would have to be careful.. certain tests have expectations of the error stack (evp_test for example)

**Comment by richsalz** [comment]
Any possibility of getting regular automated scans from parfait?

**Comment by richsalz** [comment]
See, for example issues #6969 and #6968

**Comment by slontis** [comment]
> Any possibility of getting regular automated scans from parfait?

Not anytime soon. It needs a  server.. Like other static analyzers it has false positives. I havent played around with it enough to know what happens when code shifts either

**Comment by paulidale** [comment]
Regular parfait runs was a _to get done when we've got time_ item for 3-4 years.

**Comment by paulidale** [comment]
Merged.

**Pull request closed without merging**